src/xdocs/checks/whitespace/genericwhitespace.xml

```diff
@@ -49,34 +49,66 @@
           Examples with correct spacing:
         </p>
         <source>
-// Generic methods definitions
-public void &lt;K, V extends Number&gt; boolean foo(K, V) {}
-// Generic type definition
-class name&lt;T1, T2, ..., Tn&gt; {}
-// Generic type reference
-OrderedPair&lt;String, Box&lt;Integer&gt;&gt; p;
-// Generic preceded method name
-boolean same = Util.&lt;Integer, String&gt;compare(p1, p2);
-// Diamond operator
-Pair&lt;Integer, String&gt; p1 = new Pair&lt;&gt;(1, &quot;apple&quot;);
-// Method reference
-List&lt;T&gt; list = ImmutableList.Builder&lt;T&gt;::new;
-// Method reference
-sort(list, Comparable::&lt;String&gt;compareTo);
-// Constructor call
-MyClass obj = new &lt;String&gt;MyClass();
+public class Example1 {
+
+  // Generic methods definitions
+  public static &lt;K, V&gt; boolean foo(K key, V value) {
+    // Implementation details
+    return false;
+  }
+
+  // Generic type definition
+  class name&lt;T1, T2&gt; {
+    // Implementation details
+  }
+
+  public void examples() {
+    // Generic type reference
+    Map&lt;String, List&lt;Integer&gt;&gt; p;
+
+    // Generic preceded method name
+    boolean same = Example1.&lt;Object, Object&gt;foo(null, null);
+
+    // Diamond operator
+    Map&lt;Integer, String&gt; p1;
+
+    // Method reference
+    Collections.sort(new ArrayList&lt;String&gt;(), new Comparator&lt;String&gt;() {
```

Comment:
We lost method reference before template `::<String>`.

Please restore it, code can be any but there should be `::` before generic.

Reply:
done

Reply:
done

Reply:
@chrccl, do not resolve review items, just reply "done"